src/cilium_manifests.py

```diff
@@ -1,9 +1,99 @@
 """Implementation of Cilium Manifests manager."""
 
+import logging
 from typing import Dict
 
 from ops.manifests import ConfigRegistry, ManifestLabel, Manifests, Patch
 
+log = logging.getLogger(__name__)
+
+
+class PatchHubbleMetricsConfigMap(Patch):
+    """Configure Hubble Prometheus metrics."""
+
+    def __call__(self, obj) -> None:
+        """Update hubble-metrics entry in cilium-config ConfigMap."""
+        if not (obj.kind == "ConfigMap" and obj.metadata.name == "cilium-config"):
+            return
+
+        log.info(f"Patching hubble_metrics: {self.manifests.hubble_metrics}")
+
+        if not self.manifests.hubble_metrics:
+            return
+
+        data = obj.data
+        values = {
+            "hubble-metrics": " ".join(self.manifests.hubble_metrics),
+            "hubble-metrics-server": ":9965",
+        }
+        data.update(values)
+        log.info(f"Patching Hubble metrics [{self.manifests.hubble_metrics}]: {data}")
+
+
+class PatchCiliumOperatorAnnotations(Patch):
+    """Configure Cilium-Operatior metrics expose."""
+
+    def __call__(self, obj) -> None:
+        """Update CIlium Operator Prometheus annotations."""
+        if not (obj.kind == "Deployment" and obj.metadata.name == "cilium-operator"):
+            return
+        if not self.manifests.config["enable-cilium-metrics"]:
+            return
+
+        annotations = {
+            "prometheus.io/port": "9963",
+            "prometheus.io/scrape": "true",
+        }
+
+        metadata = obj.spec.template.metadata
+        log.info(f"Metadata cilium-operator: {metadata}")
+        metadata.annotations = annotations
+        log.info(f"Metadata cilium-operator Patched: {metadata.annotations}")
+
+
+class PatchCiliumDaemonSetAnnotations(Patch):
+    """Configure Cilium DaemonSet metrics expose."""
+
+    def __call__(self, obj) -> None:
+        """Update Cilium Prometheus annotations."""
+        if not (obj.kind == "DaemonSet" and obj.metadata.name == "cilium"):
+            return
+        if not self.manifests.config["enable-cilium-metrics"]:
+            return
+
+        annotations = {
+            "prometheus.io/port": "9962",
+            "prometheus.io/scrape": "true",
+        }
+        metadata = obj.spec.template.metadata
+        log.info(f"Metadata: {metadata}")
+
+        metadata.annotations = annotations
+        log.info(f"Metadata annotatd: {metadata}")
+
+
+class PatchPrometheusConfigMap(Patch):
+    """Configure Cilium Prometheus metrics."""
+
+    def __call__(self, obj) -> None:
+        """Update Cilium Components."""
+        if not (obj.kind == "ConfigMap" and obj.metadata.name == "cilium-config"):
+            return
+
+        if not self.manifests.config["enable-cilium-metrics"]:
+            return
+
+        log.info("Patching Cilium ConfigMap Prometheus Values.")
+        values = {
+            "prometheus-serve-addr": ":9962",
```

Comment:
magic addresses?  what can you tell me about these arguments

Reply:
This configuration option is extracted directly from the Helm rendered template. It doesn't have an IP/host because it should bind to all network interfaces in the Pod. TBH, it looks weird.